corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -606,6 +613,33 @@ var dataDictionaryModel = function (dataUrl, casePropertyUrl, typeChoices, fhirR
         }
     };
 
+    self.showCasePropertyWarningIfNeeded = function () {
+        let caseType = self.getActiveCaseType();
+        if (caseType.propertyCount > self.casePropertyLimit) {
+            let content = gettext(
+                "The '" + caseType.name + "' case has " + caseType.propertyCount + " properties. " +
+                "We recommend keeping case properties below " + self.casePropertyLimit + ", " +
+                "otherwise you may run into performance issues at the time of data collection and analysis."
+            );
+            self.casePropertyWarningContent(content);
+            self.showCasePropertyWarning(true);
+        } else {
+            self.showCasePropertyWarning(false);
+        }
+    };
+
+    self.toggleCasePropertyWarning = function () {
+        let toggle = document.getElementById("performance-warning-toggle");
+        let icon = toggle.querySelector('i');
+        if (icon.classList.contains('fa-chevron-down')) {
+            icon.classList.remove('fa-chevron-down');
+            icon.classList.add('fa-chevron-up');
```

Comment:
Does changing the class on the icon trigger expand/collapse? What prevents the chevron direction from getting out of sync with expanded/collapsed state?

Should `toggleCasePropertyWarning` change the value of `showCasePropertyWarning`, or does it do that through some magical convolution that I'm missing?

If there is an issue here, there also may be an issue [over here](https://github.com/dimagi/commcare-hq/pull/36588/commits/1e1b2e0e08a26f8535d831c7edd4a23e92f1713f#diff-e511cabdd447893388e1a65c371c787f2f7c94e76697607c307131e0cdb587a5).

Reply:
+1

Reply:
I ended up using the `collapsed` class on the toggle, which I confirmed gets added and removed automatically. The hack is that I needed to explicitly add it so that it matches that starting state of the element.

[f845770](https://github.com/dimagi/commcare-hq/pull/36588/commits/f8457704c85ce848d24b3db2e335ac067773edad)